src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PathSeqBwaSpark.java

```diff
@@ -0,0 +1,142 @@
+        "order to increase parallelism.",
+        oneLineSummary = "Aligns reads to reference using Bwa-mem",
+        programGroup = SparkProgramGroup.class)
+public final class PathSeqBwaSpark extends GATKSparkTool {
+
+    private static final long serialVersionUID = 1L;
+
+    @Argument(doc = "Base uri for the output file(s). Paired and unpaired reads will be written to uri appended with" +
+            " '.paired.bam' and '.unpaired.bam'",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    public String ouputUri;
+
+    @Argument(doc = "URI to paired reads BAM",
+            fullName = "pairedInput",
+            optional = true)
+    public String inputPaired = null;
+
+    @Argument(doc = "URI to unpaired reads BAM",
+            fullName = "unpairedInput",
+            optional = true)
+    public String inputUnpaired = null;
+
+    @ArgumentCollection
+    public PSBwaArgumentCollection bwaArgs = new PSBwaArgumentCollection();
+
+    /**
+     * Reads bam from path and returns tuple of the header and reads RDD
+     */
+    private Tuple2<SAMFileHeader, JavaRDD<GATKRead>> loadBam(final String path, final PipelineOptions options,
+                                                             final ReadsSparkSource readsSource) {
+        if (path == null) return null;
+        if (BucketUtils.fileExists(path)) {
+            final SAMFileHeader header = readsSource.getHeader(path, null);
+            if (header.getSequenceDictionary() != null && !header.getSequenceDictionary().isEmpty()) {
+                throw new UserException.BadInput("Input BAM should be unaligned, but found one or more sequences in the header.");
+            }
+            final JavaRDD<GATKRead> reads = readsSource.getParallelReads(path, null, null, bamPartitionSplitSize);
+            PSBwaUtils.addReferenceSequencesToHeader(header, bwaArgs.referencePath, getReferenceWindowFunction(), options);
+            return new Tuple2<>(header, reads);
+        }
+        logger.warn("Could not find file " + path + ". Skipping...");
+        return null;
+    }
+
+    /**
+     * Writes RDD of reads to path. Note writeReads() is not used because there are separate paired/unpaired outputs.
+     * Header sequence dictionary is reduced to only those that were aligned to.
+     */
+    private void writeBam(final JavaRDD<GATKRead> reads, final String inputBamPath, final boolean isPaired,
+                          final JavaSparkContext ctx, final SAMFileHeader header) {
+
+        //Only retain header sequences that were aligned to
+        reads.cache();
+        final List<String> usedSequences = PSBwaUtils.getAlignedSequenceNames(reads);
+        final List<SAMSequenceRecord> usedSequenceRecords = usedSequences.stream()
+                .map(seqName -> header.getSequence(seqName))
+                .filter(Objects::nonNull)
```

Comment:
I hope not. The nonNull is there in case header.getSequence() returns null, which should only happen if a read aligned to a sequence that is not in the input BAM header or reference dictionary. In theory, this could happen if someone deleted header sequences from the input BAM after aligning it. Again, the tool currently forbids this but we may want to allow pre-aligned inputs in the future.

I've added a check to print a warning message if one comes up null so at least we know something is up.